jenkins/whitespace-checker.sh

```diff
@@ -0,0 +1,32 @@
+#! /usr/bin/env bash
+#
+# Copyright (c) 2019      Hewlett Packard Enterprise. All Rights Reserved.
+#
+# Additional copyrights may follow
+#
+# Check for white space violation in a given commit range.
+# Run on a PR to check whether it is introducing bad white space.
+#
+
+context=3
+config_file=.whitespace-checker-config.txt
+if [[ -r $config_file ]]; then
+    exclude_dirs=`cat $config_file`
+else
+    exclude_dirs='hwloc|libevent|pmix4x|treematch|romio'
```

Comment:
Minor note: "hwloc" will exclude the entire `opal/mca/hwloc` tree.  Can we make it a little more specific, perhaps `opal/mcal/hwloc/hwloc.*/hwloc`, or somesuch?